### PR TITLE
Float filter works with CORS short baseline test data.

### PR DESCRIPTION
**Pull request by akleeman**
- The short baseline CORS files found some (anticipated) edge cases mostly dealing with support for continuation lines when parsing the observation type and PRNs.  The files also contain sporadic comments so the parser needed to be capable of detecting and skipping those lines.
- The simple float filter (filters.KalmanFilter) now works with the CORS data (at least until satellites are added/dropped).
- Fixed the datetime64 nanosecond precision issue by dealing with time of flight directly.
- Updated synthetic data to closer align with CORS.
- Tightened thresholds on several regression-style unit tests.

### CODE REVIEW

File: gnss_analysis/dgnss.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def make_propagated_single_differences(rover, base, base_pos_ecef):
   #   precision when generating synthetic observations adds too much
   #   error to be able to discern if this is true or not.
   prop_base = propagate.delta_tof_propagate(base_pos_ecef, base,
-                                            new_toa=rover['time'].values)
+                                            new_tot=rover['tot'].values)
```
Comment (benjamin0) [inline gnss_analysis/dgnss.py]: Hmm interesting, I'm trying to reason about this. Might be something to come back to later.

File: gnss_analysis/dgnss.py [hunk 2]
```diff
@@ -101,12 +102,14 @@ def create_single_difference_objects(sdiffs):
   """
   assert sdiffs.index.name == 'sid'
   for sid, sdiff in sdiffs.iterrows():
+    if isinstance(sid, basestring) and sid.startswith('G'):
```
Comment (benjamin0) [inline gnss_analysis/dgnss.py]: maybe want to raise a NotImplemented yet for GLONASS, etc.?

File: gnss_analysis/io/rinex.py [hunk 3]
```diff
@@ -212,26 +234,43 @@ def parse_header(f):
...
-      break
-    # splits apart the line type and line content
-    content, line_type = content_parser(line)
-    # remove stray characters around the line type
-    line_type = line_type.strip()
+
+  def iter_type_and_content():
+    for line in f:
+      # the last line of the header will end with "END OF HEADER"
+      if "END OF HEADER" in line:
```
Comment (benjamin0) [inline gnss_analysis/io/rinex.py]: could use .endswith()

File: gnss_analysis/dgnss.py (hunk 2 above)
Comment (akleeman) [inline gnss_analysis/dgnss.py]: done

File: gnss_analysis/time_utils.py [hunk 4]
```diff
@@ -5,11 +5,50 @@
...
+    A floating point representation of the number of seconds
+
+  Returns
+  -----------
+  td : np.timedelta64
+    A timedelta64[ns] object (or array like of them)
+  """
   seconds = np.array(seconds)
-  return (seconds * 1e9).astype('timedelta64[ns]')
+  return np.round(seconds * 1e9).astype('timedelta64[ns]')
```
Comment (benjamin0) [inline gnss_analysis/time_utils.py]: should we do away with timedeltas so we can handle sub ns precision?

**Comment by benjamin0** [comment]
looks great!

File: gnss_analysis/time_utils.py (hunk 4 above)
Comment (akleeman) [inline gnss_analysis/time_utils.py]: Yeah I think that's a tricky question.  I switched to using `datetime64` times (instead of `wn`and `tow`) since I noticed that RINEX reported epochs using year/month/day/etc, and dealing with time that actually consisted of two variables was a pain (and likely to result in accidental inconsistencies).

The problem of course, is that `datetime64` only (at least for our purposes) supports nanosecond precision, but I think that's OK.  sbp ([for example](https://github.com/swift-nav/piksi_firmware/blob/master/src/sbp_utils.c#L65)) also only supports nanoseconds and RINEX only provides 1e-7 second resolution on epoch times.  We'll just need to be careful that internally if we're dealing with quantities that require higher precision (time of flight for example) we do so using float numbers of seconds.

File: gnss_analysis/io/rinex.py (hunk 3 above)
Comment (akleeman) [inline gnss_analysis/io/rinex.py]: done.  Had to change it to `line.strip().endswith('END OF HEADER')` ... but still probably better than what I had which would fail if somebody wrote an unfortunate comment that included `"END OF HEADER"`.
